### PR TITLE
feat: Implement HAPI17 support

### PR DESCRIPTION
**Pull request by Romakita**
BREAKING CHANGE: This plugin doesn't support HAPI v16. Callback are removed in favor of promise.

### CODE REVIEW

**Comment by Romakita** [comment]
Hello @franciscogouveia 
Thanks for your plugin and your work :)
See you
Romain

**Comment by Romakita** [comment]
Hello @franciscogouveia 
Thanks for you work :D
I fixed issues with the travis-ci configuration. Can you take a time to review this PR, please :) ? 
I used your plugins in big project and it should be great to have a new version that support HAPI17

See you
Romain

**Comment by yadomi** [comment]
Any news on this ?

**Comment by nxmohamad** [comment]
@franciscogouveia

File: lib/index.js [hunk 1]
```diff
@@ -69,70 +79,68 @@ defaults.options = {
...
 
-    next();
+        server.ext('onPostAuth', internals.retrievePolicyHandler(dataRetriever, options));
+    }
 };
 
-exports.register.attributes = {
+
+exports.plugin.register.attributes = {
     pkg: require('../package.json')
```
Comment (franciscogouveia) [inline lib/index.js]: Can use the const Pack created above
```suggestion
    pkg: Pack
```

File: test/helpers/run.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+const {createServer} = require('./server');
```
Comment (franciscogouveia) [inline test/helpers/run.js]: I believe this file is unnecessary

**Comment by franciscogouveia** [comment]
Sorry guys it took so long, I've been a bit too busy. Thank you very much for this contribution!

**Pull request merged**